lib/util/src/map.c

```diff
@@ -200,6 +222,10 @@ map_lookup(struct map *map, uint64_t key, uintptr_t *val_out)
     return found;
 }
 
+// Remove a key from a map and optionally return its value
+// Return:
+// - true : key found and removed, value returned in val if val!=NULL
+// - false : key not found
```

Comment:
Would rather we don't use the single-line comment form for multiline comments...